test/www/jxcore/test/thalicryptomanagertests.js

```diff
@@ -0,0 +1,130 @@
+"use strict";
+
+require('./mockmobile');
+var fs = require('fs-extra-promise');
+var path = require('path');
+var crypto = require('crypto');
+var tape = require('wrapping-tape');
+var cryptomanager = require('../thali/thalicryptomanager');
+
+// get the values needed for running the tests
+var configValues = cryptomanager.getConfigValuesForTestingOnly();
+
+var fileLocation = './pkcs12folder';
+
+// test setup & teardown activities
+var test = tape({
+  setup: function(t) {
+    fs.mkdirSync(fileLocation);
+    t.end();
+  },
+  teardown: function(t) {
+    fs.remove(fileLocation, function(err) {
+      if(err) {
+        console.log('could not remove folder - err: ', err);
+      }
+      t.end();
+    });
+  }
+});
+
+test('successfully create a new pkcs12 file and return hash value', 2,
+  function(t) {
+  var errorMessage = null;
+  
+  Mobile.storeGetDocumentsPathReturnArguments(errorMessage, fileLocation);
+  
+  cryptomanager.getPublicKeyHash(function (err, publicKeyHash) {
+    t.equal(err, null);
+
+    var file = path.join(fileLocation, configValues.pkcs12FileName);
+    fs.readFile(file, function (err, pkcs12Content) {
+      if(err) {
```

Comment:
I'm confused by lines 42 through 45. Aren't your effectively just re-inventing t.error? I am assuming that when you use a framework you go read all the instructions about that framework. So didn't you go to https://www.npmjs.com/package/tape and look at what methods it supports and see which ones already do what you need?